mlir/lib/Driver/CompilerDriver.cpp

```diff
@@ -258,6 +264,14 @@ OwningOpRef<ModuleOp> parseMLIRSource(MLIRContext *ctx, const llvm::SourceMgr &s
     return parseSourceFile<ModuleOp>(sourceMgr, parserConfig);
 }
 
+/// From the MLIR module it checks if gradients operations are in the program.
+bool containsGradients(mlir::ModuleOp moduleOp)
+{
+    bool contain = false;
+    moduleOp.walk([&](catalyst::gradient::GradientOpInterface op) { contain = true; });
```

Comment:
This can be made more efficient by using the interrupt walker, allowing for early exit as soon as we found one gradient operation.

Reply:
Done